src/css/hiddenVisibleSelectors.js

```diff
@@ -4,12 +4,10 @@ define([
 ], function( jQuery ) {
 
 jQuery.expr.filters.hidden = function( elem ) {
-	// Use OR instead of AND as the element is not visible if either is true
-	// See tickets #10406 and #13132
-	return !elem.offsetWidth || !elem.offsetHeight;
+	return !jQuery.expr.filters.visible( elem );
```

Comment:
We haven't noticed a perf loss and this way guarantees hidden and visible are opposites.